config/log_config_test.go

```diff
@@ -0,0 +1,90 @@
+package config
+
+import (
+	"testing"
+
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/logging"
+)
+
+func TestValidatePattern(t *testing.T) {
+	test.That(t, validatePattern("robot_server.resource_manager"), test.ShouldBeTrue)
+	test.That(t, validatePattern("robot_server.resource_manager.*"), test.ShouldBeTrue)
+	test.That(t, validatePattern("robot_server.*.resource_manager"), test.ShouldBeTrue)
+	test.That(t, validatePattern("robot_server.*.*"), test.ShouldBeTrue)
+	test.That(t, validatePattern("*.resource_manager"), test.ShouldBeTrue)
+	test.That(t, validatePattern("*"), test.ShouldBeTrue)
+
+	test.That(t, validatePattern("robot_server..resource_manager"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.resource_manager."), test.ShouldBeFalse)
+	test.That(t, validatePattern(".robot_server.resource_manager"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.resource_manager.**"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.**.resource_manager"), test.ShouldBeFalse)
+
+	test.That(t, validatePattern("_.robot_server.resource_manager"), test.ShouldBeFalse)
+	test.That(t, validatePattern("-.robot_server"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.-"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.-"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server.-.resource_manager"), test.ShouldBeFalse)
+	test.That(t, validatePattern("robot_server._.resource_manager"), test.ShouldBeFalse)
+}
+
+func TestUpdateLoggerRegistry(t *testing.T) {
```

Comment:
let's add a case with a wildcard not at the end - e.g. `rdk.*.modmanager`

Reply:
there is a test for that at L14, testing `robot_server.*.resource_manager`

Reply:
we should have one in `TestUpdateLoggerRegistry`

Reply:
ah I see, working on it